### PR TITLE
let codes follow ES2015

### PR DESCRIPTION
**Pull request by hiroppy**
Hello :)
## change of the syntax
- `Arrow functions`
- `Template strings`
- `variable declaration`
## Edit range
- `build.js`
- `scripts/`
- `events/`
## Check nodejs versions
- [x] 4.2.2
- [x] 5.9.1

Cheers

### CODE REVIEW

**Comment by ghost** [comment]
[Travis build passed](https://travis-ci.org/nodejs/nodejs.org/builds/118107855) :+1:

**Comment by MylesBorins** [comment]
@abouthiroppy did you verify that all of the instances of const are for objects not being mutated? If you just did a regex or a find and replace there might be some bugs in there

**Comment by hiroppy** [comment]
@TheAlphaNerd  I checked `npm run build`, `npm load-version`, `npm test` and `node *.js`. 
So, there might be a danger. Is `let` all right?

**Comment by MylesBorins** [comment]
Yes and no. We recently found a bunch of problems in core with let being up to 3x slower in a for loop. That is generally only a problem in hot code.

We tend to be stricter on core than in this project though.  @fhemberger do you have any thoughts?

**Comment by stevemao** [comment]
CC @nodejs/documentation

**Comment by fhemberger** [comment]
Personally, I'm already using ES6 all over the place (including `let` and `const`). I'm not really working with time critical/hot code most of the time, so even if `let` is still slower at the moment, that's no deal breaker for me. Upcoming versions of V8 (and other engines as well) will improve on that particular case as well. Also regarding the build process of the website, I don't think there will be a measurable impact at all.

@phillipj may want to take a look as well, as many of the scripts were written by him.

**Comment by ghost** [comment]
i do like this a lot. the only thing i'd like to see would be an information under the `// BUILD.JS: etc` comments that points out that this is, in fact, ES6 code.

**Comment by fhemberger** [comment]
I don't see a need for that. ES6 is the current official standard of the JavaScript language.

**Comment by ghost** [comment]
@fhemberger true

File: build.js [hunk 1]
```diff
@@ -44,12 +44,12 @@ const markedOptions = {
...
-  var defaultJSON = require(`./locale/${DEFAULT_LANG}/site.json`)
-  var templateJSON = require(`./locale/${lang}/site.json`)
-  var finalJSON = JSON.parse(JSON.stringify(defaultJSON))
-  var merge = function (targetJSON, customJSON) {
-    Object.keys(customJSON).forEach(function (key) {
-      let value = customJSON[key]
+  const defaultJSON = require(`./locale/${DEFAULT_LANG}/site.json`)
+  const templateJSON = require(`./locale/${lang}/site.json`)
+  const finalJSON = JSON.parse(JSON.stringify(defaultJSON))
+  const merge = (targetJSON, customJSON) => {
```
Comment (lpinca) [inline build.js]: Is deep clone required here? If not we can use `Object.assign()`.

Edit: It doesn't seem to be required, the cloned object is only passed to `metalsmith.metadata` so I guess it's ok to use `Object.assign()`.

File: build.js [hunk 2]
```diff
@@ -208,20 +208,20 @@ function buildLocale (source, locale) {
...
-    Object.keys(files).forEach(function (path) {
+    Object.keys(files).forEach((path) => {
       if (!isEditable.test(path)) {
         return
       }
 
-      var file = files[path]
-      var url = 'https://github.com/nodejs/nodejs.org/edit/master/locale/' + options.locale + '/' + path.replace('.html', '.md')
+      const file = files[path]
+      const url = 'https://github.com/nodejs/nodejs.org/edit/master/locale/' + options.locale + '/' + path.replace('.html', '.md')
```
Comment (lpinca) [inline build.js]: I would use template string here for consistency.

File: events/pull-nodeschool.js [hunk 3]
```diff
@@ -28,11 +31,11 @@ request('http://nodeschool.io/chapters/list.json', {json: true}, function (e, re
   })
 
   function _geo () {
-    if (chapters.length === 0) return yml.save()
-    var chapter = chapters.shift()
-    geocoder.geocode(chapter.location, function (err, res) {
+    if (chapters.length === 0) { return yml.save() }
```
Comment (lpinca) [inline events/pull-nodeschool.js:34]: Does linter complain about curly braces?

**Comment by lpinca** [comment]
Regarding arrow functions, are round brackets required in our style when there is a single argument?

**Comment by eljefedelrodeodeljefe** [comment]
The discssion popped up every now and then and I think concsensus was that they are required always, even with no argument. See [process](https://nodejs.org/api/process.html) for reference.

**Comment by lpinca** [comment]
@eljefedelrodeodeljefe yeah I noticed that core always uses round brackets, there is also a linter rule for it. I guess we should do the same then.
AFAIK Standard (which is what we are using here) does not enforce it.

File: events/pull-nodeschool.js (hunk 3 above)
Comment (hiroppy) [inline events/pull-nodeschool.js:34]: No, but curly brace is written in other files.

Comment (lpinca) [inline events/pull-nodeschool.js:34]: You are right, there isn't a consistent style for this, not even in this diff.
Forget the above comment.

**Comment by lpinca** [comment]
LGTM.

**Comment by hiroppy** [comment]
I also think we should attach a round bracket.

File: build.js (hunk 1 above)
Comment (phillipj) [inline build.js]: I'd also like to see `Object.assign()` here. I'm okey with that change being a separate PR though.

**Comment by phillipj** [comment]
LGTM

@fhemberger I completely agree with you about performance not being an issue for us.

File: build.js (hunk 1 above)
Comment (hiroppy) [inline build.js]: Oh, I changed it here. Thanks:)

File: build.js [hunk 4]
```diff
@@ -44,28 +44,18 @@ const markedOptions = {
...
-      }
-    })
-  }
-  merge(finalJSON, templateJSON)
-  return finalJSON
+  const defaultJSON = require(`./locale/${DEFAULT_LANG}/site.json`)
+  const templateJSON = require(`./locale/${lang}/site.json`)
+  const finalJSON = JSON.parse(JSON.stringify(defaultJSON))
+
+  return Object.assign(finalJSON, templateJSON)
```
Comment (lpinca) [inline build.js]: What about:

``` js
return Object.assign({}, defaultJSON, templateJSON)
```

so we can get rid of clone via stringify/parse.

**Comment by lpinca** [comment]
:shipit: ?

**Comment by hiroppy** [comment]
Thank you for many reviews and opinions😊

**Comment by ghost** [comment]
[Travis build passed](https://travis-ci.org/nodejs/nodejs.org/builds/118288644) :+1:

File: build.js (hunk 4 above)
Comment (phillipj) [inline build.js]: @abouthiroppy ^^ notice that @lpinca removed `finalJSON` completely

File: build.js [hunk 5]
```diff
@@ -44,28 +44,18 @@ const markedOptions = {
...
-      } else {
-        targetJSON[key] = value
-      }
-    })
-  }
-  merge(finalJSON, templateJSON)
-  return finalJSON
+  const defaultJSON = require(`./locale/${DEFAULT_LANG}/site.json`)
+  const templateJSON = require(`./locale/${lang}/site.json`)
+  const finalJSON = JSON.parse(JSON.stringify(defaultJSON))
```
Comment (lpinca) [inline build.js]: As noted by @phillipj there is still one nit. Remove this line, and use `defaultJSON` in `Object.assign`:

``` js
return Object.assign({}, defaultJSON, templateJSON)
```

Thanks!

Comment (hiroppy) [inline build.js]: Yeah, definitely. We didn't need `JSON.parse(JSON.stringify(obj))` for deep copy.
I will modify it:)

**Pull request merged**